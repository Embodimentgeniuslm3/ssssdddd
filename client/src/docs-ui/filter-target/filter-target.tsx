import {Component, Host, Prop, h, Watch, State} from "@stencil/core";
import {pageContext} from "../page/page.context";
import {SelectedFilters} from "../page/page.types";

@Component({tag: "docs-filter-target", shadow: false})
export class DocsFilterTarget {
  /*** the currently-selected filter state */
  @Prop() readonly selectedFilters?: SelectedFilters;
  /*** the conditions off of which to style the host visible vs. hidden */
  @Prop() readonly filters?: Record<string, string>;

  @State() shouldDisplay = false;

  @Watch("filters")
  @Watch("selectedFilters")
  componentWillLoad() {
    if (!this.selectedFilters) {
      this.shouldDisplay = false;
      return;
    }

    if (this.filters && this.selectedFilters) {
      for (const [filterKey, filterValue] of Object.entries(this.filters)) {
        if (this.selectedFilters[filterKey] !== filterValue) {
          this.shouldDisplay = false;
          return;
        }
      }
    }

    this.shouldDisplay = true;
  }

  render() {
    return (
      <Host style={{display: this.shouldDisplay ? "initial" : "none"}}>
        <slot name="content" />
      </Host>
    );
  }
}

pageContext.injectProps(DocsFilterTarget, ["selectedFilters"]);
