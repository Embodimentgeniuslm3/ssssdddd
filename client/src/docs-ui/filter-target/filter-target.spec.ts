import {DocsFilterTarget} from "./filter-target";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-filter-target", () => {
  describe("Component logic", () => {
    let filterTarget: DocsFilterTarget;
    beforeEach(() => (filterTarget = new DocsFilterTarget()));

    it("should init `selectedFilters` as `undefined`", () =>
      expect(filterTarget.selectedFilters).toBe(undefined));
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsFilterTarget],
            html: `<docs-filter-target />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
