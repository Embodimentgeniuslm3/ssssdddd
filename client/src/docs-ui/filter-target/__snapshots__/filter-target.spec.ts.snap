// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-filter-target Render logic should render 1`] = `
<docs-filter-target style="display: none;">
  <!---->
</docs-filter-target>
`;
